src/kafka-util/src/bin/kgen.rs

```diff
@@ -271,6 +280,19 @@ impl<'a> RandomAvroGenerator<'a> {
                 v.extend(iter::repeat_with(sample).take(len));
             }
         }
+        fn decimal_dist(
+            json: &serde_json::Value,
+            rng: Rc<RefCell<ThreadRng>>,
+            precision: usize,
+        ) -> impl FnMut() -> Vec<u8> {
+            let x = json.as_array().unwrap();
+            let (min, max): (i64, i64) = (x[0].as_i64().unwrap(), x[1].as_i64().unwrap());
+            // Ensure values fit within precision bounds.
+            assert!(10i64.pow(u32::try_from(precision).unwrap()) > max);
+            assert!(10i64.pow(u32::try_from(precision).unwrap()) > min.abs());
```

Comment:
what happens if 10^precision is larger than what can fit in an i64?

Reply:
Fixed with `checked_pow`